### DISCUSSION TITLE
How to submit a new entry

### DISCUSSION THREAD

**Discussion by nelsonmestevao** [cesium/CeSIUMpeloMundo#26] [Ideas]
At the moment, everyone is free to open a pull request and submit a new entry by adding it to `data/places.json` and the photo to `public/photos/{github username}/{city}.jpg`. 

This approach as the following advantages: 

- it's simple, we just do it in a clear process that we understand - GH Pull Requests
- the maintainers can dismiss non-fitting contributions (like bots or inappropriate photos) by closing the PR
- no need to manage authentication or authorization on our side, we just use GitHub accounts (also good for the future because we don't need to keep one more admin account around)
- easy deployments and deployment previews, all the content is already on the repository, no need for bucket storage or API calls
- we can keep track of not only code contributions over time in the GitHub contributors page

However, this as the following disadvantages:

- the repository will keep growing in size and could be a problem one day

### 2. What are your ideas for other systems (in terms of design flow and implementation details)?

**Comment by diogogmatos** [comment]
### Here's the main disadvantages I think we saw with the current implementation:

- A GitHub PR assumes that the user knows the data structure of the project and demands that they manually edit the code;

- The lack of a graphical interface makes it harder for anyone to easily submit a request for a new marker. For example, not enabling them to submit from a mobile device where the picture is usually stored.

### Here's a possibilty we discussed:

Making use of a third party system to submit a new entry, like a sort of Google Forms, avoiding having to code a proprietary interface for it. The idea would be to add a bridge between the submission and the pull request, where someone would still need to manually add the new marker to the frontend.

**Advantages:**

- No need for an authentication system, the user can identify himself through the submission form;

- Users unfamiliar with the projects structure can still submit a request for a new marker;

- A request can be *easily* done from any device with an internet connection;

- A link for new submissions can be shared on social media, potentially attracting more people to the project.

**Disadvantages:**

- Someone needs to be responsible for reading, evaluating and eventually adding the new entries to the project, instead of it being done by the user itself;

- No connection between the submissions interface and the projects code.

### In conclusion:

The ultimate goal would be to eliminate the need to manually edit the code, having the submissions interface integrate with the project and update the database (while still requiring someone to approve the requests).

Tell me what you think! Is it dumb? Is it unnecessary? It might be xD.

**Reply by nelsonmestevao** [reply]
I think the Google Forms is not as good as the current one because:

- when submitting files through a Google form, you need to authenticate yourself (requiring the user to have a Google account)
- although the person who is submitting a new entry can do it from their phone it will not make it faster, because someone from the admin team will have to keep track of this new form submission system and do it manually (it doesn't scale)
- the person adding a submission can not see a preview of the result (right now, you just open the PR, and automatically you get a link with the deployment preview) which gives the ability to the person to fix or improve the new pin submission even before any review
- the user does not get any feedback on approvals, requests for changes or when the submission is denied
- fewer people would be able to help maintain because they would have to have also access to the form submissions
- the one contributing would not get into the contributors of this repository and other people would get the "credit"
- contributors lose the opportunity to learn more about contributing to open source projects and dealing with git and GitHub

The current implementation does not require the user to change code. Only the entries file, and we could and should add detailed instructions in the README to make it easier. We should also create some criteria, for example, limit the resolution of pictures.

**Reply by diogogmatos** [reply]
Your totally right, we should just update the README so that anyone can easily submit a new entry. Thanks for the feedback!

**Reply by FranciscoLira** [reply]
I would love the submissions to be done directly on the website, but for that we would need a lot of other things. But I think that is the way to go. And with the possibility of the CeSIUM ID, this could be just one more platform to have it. 

In the meantime, I think a good README would be the best option. With all detailed steps.
